pymeasure/instruments/keithley/keithley2602B.py

```diff
@@ -0,0 +1,212 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.keithley.keithley2600 import Keithley2600
+from pymeasure.instruments.validators import strict_discrete_set, truncated_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class Keithley2602B(Keithley2600):
+    """Represents the Keithley 2602B SourceMeter. This class adds digital I/O
+    functionality to the Keithley2600 series driver. Note that this driver can
+    be used to control any of the other 2600 series models that support digital
+    I/O (2601B, 2611B, 2612B, 2635B, 2636B).
+    """
+
+    number_of_pins = 14
+    _event_descriptions = []
+
+    for channel in ["a", "b"]:
+        _event_descriptions += [
+            f"smu{channel}.trigger.SOURCE_COMPLETE_EVENT_ID",
+            f"smu{channel}.trigger.MEASURE_COMPLETE_EVENT_ID",
+            f"smu{channel}.trigger.PULSE_COMPLETE_EVENT_ID",
+            f"smu{channel}.trigger.SWEEP_COMPLETE_EVENT_ID",
+            f"smu{channel}.trigger.IDLE_EVENT_ID",
+        ]
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(adapter, includeSCPI=False, **kwargs)
+
+        self.dio_pins = [
+            Keithley2600DigitalIOPin(self, i + 1) for i in range(self.number_of_pins)
+        ]
+
+    @staticmethod
+    def get_trigger_event_description_strings():
+        """Returns a list of the valid event IDs that can be used to select the event that
+        causes a trigger to be asserted on the digital output line. The list can be indexed
+        to set a digitial I/O line to assert a trigger given the described conditions. E.g.
+        to set digital line 4 to assert a trigger when the SMU completes a source
+        action on channel A,
+        use the following:
+
+        .. code-block:: python
+
+            #Assume a Keithley2602B object called "smu" has been successully instaniated
+            trigger_event_lists = smu.get_trigger_event_description_strings()
+            smu.dio_pins[3].stimulus = trigger_event_lists[0]
+
+        See page 9-61 of the Reference Manual for more details about the various event
+        descriptions.
+        """
+
+        return Keithley2602B._event_descriptions
```

Comment:
Is there a better way to return to the user the list of acceptable event description strings?

Reply:
Just make the list a public property (without the underscore), such that the user may access it via smu.event_descriptions (or with the class name in front of the dot).

I don't see the point in a static method returning an publicly accessible class variable.